tests/test_prolongation.cpp

```diff
@@ -0,0 +1,297 @@
+ *  
+ *  The Test creates an arbitrary grid-function on the coarser level and prolongates it. 
+ *  On the fine level we iterate over all nodes and test the result based on whether the node is fine in theta, r or in both.
+ *
+ *  Parametrized tests are used to test for different grid sizes and with or without Dirichlet boundary conditions.
+ */
+
+TEST_P(test_prolongation, test_bilinear_prolongation)
+{
+    //we vary the grid size to guarantee that the problem works for all sizes
+    gyro::icntl[Param::nr_exp]         = (int)(std::get<0>(GetParam()) / 3) + 3;
+    gyro::icntl[Param::ntheta_exp]     = (std::get<0>(GetParam()) % 3) + 3;
+    gyro::icntl[Param::DirBC_Interior] = std::get<1>(GetParam());
+    if (gyro::icntl[Param::nr_exp] == 3)
+        gyro::icntl[Param::fac_ani] = 2; //anisotropy should not exceed grid size
+
+    gmgpolar test_p;
+    create_grid(test_p);
+
+    level& p_level = *(test_p.v_level[0]);
+    int ctheta_int = test_p.v_level[1]->ntheta_int; //number of coarse nodes in theta direction
+
+    p_level.m  = test_p.v_level[0]->nr * test_p.v_level[0]->ntheta; //fine grid size
+    p_level.mc = test_p.v_level[1]->nr * test_p.v_level[1]->ntheta; //coarser grid size
+
+    std::vector<double> u_test(p_level.mc);
+    for (int z = 0; z < p_level.mc; z++) {
+        u_test[z] =
+            1 - z +
+            pow(PI,
+                -z * z); //constructing arbitrary grid-function on coarse level to test our prolongation operator with.
+    }
+
+    std::vector<double> sol =
+        p_level.apply_prolongation_bi(u_test); //apply prolongation operator on arbitrary grid-function
+
+    for (int j = 0; j < p_level.nr_int + 1; j++) {
+        for (int i = 0; i < p_level.ntheta_int; i++) {
+            if (j % 2 == 0 && i % 2 == 0) { //testing coarse node injection
+                EXPECT_EQ(u_test[(j / 2) * ctheta_int + (i / 2)], sol[j * p_level.ntheta_int + i])
+                    << "coarse node injection is failing";
+            }
+            else if (i % 2 != 0 && j % 2 == 0) {
+            // as numbering in angle (theta_i) is odd, we have a fine node with
+            // coarse neighbors at (r_j, theta_i - k_{i-1}) and (r_j, theta_i + k_i)
+
+                double k_qm1 = p_level.theta[i] - p_level.theta[i - 1]; //calculate k_{q-1}
+                double k_q   = (i < p_level.ntheta_int - 1) ? p_level.theta[i + 1] - p_level.theta[i]
+                                                          : 2 * PI - p_level.theta[i]; //k_q
+
+                int i1 = (j / 2) * ctheta_int + (i - 1) / 2; //bottom coarse node in theta
+                int i2 = (i < p_level.ntheta_int - 1) ? i1 + 1 : (j / 2) * ctheta_int; //top coarse node in theta
+
+                double val = (k_q * u_test[i1] + k_qm1 * u_test[i2]);
+
+                EXPECT_NEAR(val, (k_q + k_qm1) * sol[j * p_level.ntheta_int + i], 1e-8) //compare values as in the paper
```

Comment:
```suggestion
                EXPECT_NEAR(val, (k_q + k_qm1) * sol[j * p_level.ntheta_int + i], 1e-12) //compare values as in the paper
```
Can we make this succeed with 1e-12 too?